docs/advanced/settings.md

```diff
@@ -503,6 +503,61 @@ This setting can only be used if the `upgrade-checker-enabled` setting is set to
 https://your.upgrade.checker-url/v99/checkupgrade
 ```
 
+### `upgrade-config`
+
+**Definition**: Upgrade-related configuration.
+
+**Default value**: `{"imagePreloadOption":{"strategy":{"type":"sequential"}}}`
+
+**Example**:
+
+```json
+{
+  "imagePreloadOption": {
+    "strategy": {
+      "type": "parallel",
+      "concurrency": 2
+    }
+  }
+}
+```
+
+#### `imagePreloadOption`
+
+**Definition**: Options for the image-preloading phase.
+
+##### `strategy`
+
+**Definition**: Image preload strategy.
+
+###### `type`
+
+**Definition**: Type of the image preload strategy.
+
+There are three available types:
+
+- `skip`: Nodes will not preload images
+- `sequential`: Nodes will preload images one by one. This is the default.
+- `parallel`: Nodes will preload images in batches. This can be fine-tuned by the `concurrency` option.
+
+:::important
+
+Please do not use `skip` if your cluster is air-gapped. Harvester upgrade will fail if no new images are preloaded to the nodes or downloadable from any registry.
```

Comment:
Please recheck this statement.

When you choose `skip` please make sure:
(1) You have set a `container-registry` and all the required images are added to this registry in advance.

This looks to be the best match scenario for this option. And, Harvester needs to publish a full image list of each Release e.g. to the suggested `version.yaml`, to help user prepare the image.

A furhter enhancement will be: If user sets this option and clicks `Upgrade`, Harvester checks the container registry and the `version.yaml`,  make sure all images are ready.

(2) You plan to pull all images from docker hub when images are required. Be sure your cluster has a high speed internet access. And also notice the potential internet interruption or docker [increase-rate-limits](https://www.docker.com/increase-rate-limits/) .

The failure of downloading even one image may cause the failure of the whole upgrade and may leave the cluster in a middle state.

This is not recommended on production clusters.

Reply:
We already seem to have the image list published in previous releases. Adding those image names to `version.yaml` during packaging should be fairly easy. Thanks for the suggestion.